internal/armada/server/submit.go

```diff
@@ -161,23 +164,33 @@ func (server *SubmitServer) SubmitJobs(ctx context.Context, req *api.JobSubmitRe
 
 func (server *SubmitServer) CancelJobs(ctx context.Context, request *api.JobCancelRequest) (*api.CancellationResult, error) {
 	if request.JobId != "" {
-		jobs, e := server.jobRepository.GetExistingJobsByIds([]string{request.JobId})
-		if e != nil {
-			return nil, status.Errorf(codes.Internal, e.Error())
+		jobs, err := server.jobRepository.GetExistingJobsByIds([]string{request.JobId})
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, err.Error())
 		}
 		return server.cancelJobs(ctx, jobs[0].Queue, jobs)
 	}
 
 	if request.JobSetId != "" && request.Queue != "" {
-		ids, e := server.jobRepository.GetActiveJobIds(request.Queue, request.JobSetId)
-		if e != nil {
-			return nil, status.Errorf(codes.Aborted, e.Error())
+		ids, err := server.jobRepository.GetActiveJobIds(request.Queue, request.JobSetId)
+		if err != nil {
+			return nil, status.Errorf(codes.Aborted, err.Error())
 		}
-		jobs, e := server.jobRepository.GetExistingJobsByIds(ids)
-		if e != nil {
-			return nil, status.Errorf(codes.Internal, e.Error())
+
+		batches := util.Batch(ids, server.cancelJobsBatchSize)
+		cancelledIds := []string{}
+		for _, batch := range batches {
+			jobs, err := server.jobRepository.GetExistingJobsByIds(batch)
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, err.Error())
+			}
+			result, err := server.cancelJobs(ctx, request.Queue, jobs)
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, err.Error())
+			}
+			cancelledIds = append(cancelledIds, result.CancelledIds...)
```

Comment:
What happens if a jobset gets half cancelled? should we not pass this through with the error information to at least have some accountability on what got cancelled and what didnt?